lib/collection/src/hash_ring.rs

```diff
@@ -89,27 +98,33 @@ impl<T: Hash + Copy + PartialEq> HashRing<T> {
         true
     }
 
-    pub fn remove_resharding(&mut self, shard: T) -> bool
+    pub fn end_resharding(&mut self, shard: T, direction: ReshardingDirection) -> bool
     where
         T: fmt::Display,
     {
         let Self::Resharding { old, new } = self else {
-            log::warn!("removing resharding shard, but hashring is not in resharding mode");
+            log::warn!("ending resharding hashring, but it is not in resharding mode");
             return false;
         };
 
         let mut old = old.clone();
         let mut new = new.clone();
 
-        let removed_from_old = old.remove(&shard);
-        let removed_from_new = new.remove(&shard);
+        let (updated_old, updated_new) = match direction {
+            ReshardingDirection::Up => (old.remove(&shard), new.remove(&shard)),
+            ReshardingDirection::Down => {
+                old.add(shard);
+                new.add(shard);
+                (true, true)
+            }
```

Comment:
It should likely return `(false, true)` by default, because we can expect the shard to still be in the old hash ring.

In that case no errors are reported.